xarray/plot/animate.py

```diff
@@ -0,0 +1,179 @@
+    aspect : scalar, optional
+        Aspect ratio of plot, so that ``aspect * size`` gives the width in
+        inches. Only used if a ``size`` is provided.
+    size : scalar, optional
+        If provided, create a new figure for the plot with the given size.
+        Height (in inches) of each plot. See also: ``aspect``.
+    ax : matplotlib axes object, optional
+        Axis on which to plot this figure. By default, use the current axis.
+        Mutually exclusive with ``size`` and ``figsize``.
+    x, y : string, optional
+        Dimensions or coordinates for x, y axis.
+        Only one of these may be specified.
+        The other coordinate plots values from the DataArray on which this
+        plot method is called.
+    xscale, yscale : 'linear', 'symlog', 'log', 'logit', optional
+        Specifies scaling for the x- and y-axes respectively
+    xticks, yticks : Specify tick locations for x- and y-axes
+    xlim, ylim : optional
+        Specify x- and y-axes limits.
+    xincrease : None, True, or False, optional
+        Should the values on the x axes be increasing from left to right?
+        if None, use the default for the matplotlib function.
+    yincrease : None, True, or False, optional
+        Should the values on the y axes be increasing from top to bottom?
+        if None, use the default for the matplotlib function.
+    **kwargs : optional
+        Additional arguments to animatplot.blocks.Line
+
+    """
+
+    from animatplot.blocks import Line, Title
+    from animatplot.animation import Animation
+
+    row = kwargs.pop('row', None)
+    col = kwargs.pop('col', None)
+    if row or col:
+        raise NotImplementedError
+
+    hue = kwargs.pop('hue', None)
+    if hue:
+        raise NotImplementedError
+
+    _check_animate_over(darray, animate_over)
+    darray = _transpose_before_animation(darray, animate_over)
+
+    ndims = len(darray[animate_over].dims)
+    if ndims > 1:
+        raise NotImplementedError
+
+    # Ensures consistency with .plot method
+    figsize = kwargs.pop('figsize', None)
+    aspect = kwargs.pop('aspect', None)
+    size = kwargs.pop('size', None)
+    ax = kwargs.pop('ax', None)
+    hue = kwargs.pop('hue', None)
+    x = kwargs.pop('x', None)
+    y = kwargs.pop('y', None)
+    xincrease = kwargs.pop('xincrease', None)  # default needs to be None
+    yincrease = kwargs.pop('yincrease', None)
+    xscale = kwargs.pop('xscale', None)  # default needs to be None
+    yscale = kwargs.pop('yscale', None)
+    xticks = kwargs.pop('xticks', None)
+    yticks = kwargs.pop('yticks', None)
+    xlim = kwargs.pop('xlim', None)
+    ylim = kwargs.pop('ylim', None)
+    _labels = kwargs.pop('_labels', True)
+
+    ax = get_axis(figsize, size, aspect, ax)
+    xplt, yplt, hueplt, xlabel, ylabel, huelabel = \
+        _infer_line_data(darray, x, y, hue, animate_over)
+
+    # Remove pd.Intervals if contained in xplt.values.
+    if _valid_other_type(xplt.values, [pd.Interval]):
```

Comment:
Should we just move this bit into `_infer_line_data`? I think only `xplt_val` is used in the remaining part of this function.